api-call.py

```diff
@@ -1,29 +1,322 @@
+    def get_api_data(self, url: str, parameters: Dict[str, Any], header: Dict[str, str]) -> Any:
+        """Get Request for getting the data from API
+
+        :param url: URL of the API
+        :param parameters: The parameters for API
+        :param header: The headers for API with authorization
+        :return: The response data from the API
+        """
+
+        response = requests.get(
+            url,
+            params=parameters,
+            headers=header,
+            timeout=10
+        )
+        if response.status_code == 200:
+            return self.format_result(response)
+        else:
+            print(response.status_code)
+            generated_error = f'There is a {response.status_code} error with this request'
+            self.log_message(generated_error)
+
+    def format_result(self, response: requests.Response) -> Any:
+        """Returns data from response object after JSON validation
+
+        :param response: The API response object
+        :return: The validated data
+        """
+        try:
+            data = response.json()
+        except json.decoder.JSONDecodeError:
+            print("Invalid JSON response")
+
+        text = json.dumps(data, indent=4)
+        return data
+
+    def show_tasks(self) -> None:
+        """Show tasks to choose as a Menu
+        """
+        api_options = ['Quit',
+                    'Beavers Bus',
+                    'Terms',
+                    'Search Text Books of a Term by a custom/any Date',
+                    'Get Details of a Vehicle and its Stops given a specified Route, with ETA and current destination']
         
-    api_call =  get_api_data("https://api.oregonstate.edu/v1/beaverbus/routes", parameters, header)
+        print('Which API data do you want to check?')
+        for i, option in enumerate(api_options):
+            print(f'{i}. {option}')
+
+    def get_user_choice(self) -> str:
+        """Get user input from menu
+
+        :returns: The user choice
+        """
+        return input('Enter your choice (0 or 1 or 2 or 3 or 4): ')
+
+    def get_url(self, choice: str) -> Dict[str, Any]:
+        """Get URL for API
+
+        :param choice: user choice input
+        :returns: API URL Object after user choice
+        """
+
+        access_token = self.get_access_token()
+        header = {'Content-Type': 'application/x-www-form-urlencoded;charset=utf-8', 'Authorization': access_token}
+        url_obj = {}
+        if choice == '0':
+            return 'Exiting...'
+        elif choice == '1':
+            url_obj = {'url': self.config['apiUrls']['beaver_bus'],
+                    'parameters': {},
+                    'header': header}
+        elif choice == '2':
+            url_obj = {'url': self.config['apiUrls']['terms'],
+                    'parameters': {},
+                    'header': header}
+        elif choice == '3':
+            user_date = input('Enter Date (yyyy-mm-dd): ')
```

Comment:
But a user can still put an invalid date format, right?

Reply:
It returns this: 400: Bad Request - The server could not understand the request due to invalid syntax or missing parameters.